### PR TITLE
osmfilter: adding zlib-dependency

### PR DESCRIPTION
**Pull request by kkdd**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/linuxbrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/linuxbrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
- [ ] Have you included the output of `brew gist-logs <formula>` of the build failure if your PR fixes a build failure. Please quote the exact error message.

-----

### CODE REVIEW

**Comment by kkdd** [comment]
I can't resolve the following "--keep-old" issue:

==> brew bottle --merge --write --no-commit ./osmfilter--0.9.x86_64_linux.bottle.json --keep-old
Error: --keep-old was passed but there are changes in:
==> osmfilter
==> FAILED

File: Formula/osmfilter.rb [hunk 1]
```diff
@@ -7,7 +7,6 @@ class Osmfilter < Formula
   head "https://gitlab.com/osm-c-tools/osmctools.git"
 
   bottle do
-    cellar :any_skip_relocation
```
Comment (issyl0) [inline Formula/osmfilter.rb:10]: Please reinstate this line, but remove the line ending `:x86_64_linux`. That'll fix the CI error.

File: Formula/osmfilter.rb [hunk 2]
```diff
@@ -19,9 +18,11 @@ class Osmfilter < Formula
   depends_on "autoconf" => :build
   depends_on "automake" => :build
 
+  uses_from_macos "zlib"
+
   resource "pbf" do
     url "https://download.gisgraphy.com/openstreetmap/pbf/AD.tar.bz2"
-    sha256 "403d74dd62f7cc59c044965c52e0e0cb8dcf2c01faa205a668e94d7258d89ad1"
+    sha256 "f8decd915758139e8bff2fdae6102efa0dc695b9d1d64cc89a090a91576efda9"
```
Comment (issyl0) [inline Formula/osmfilter.rb:25]: Why did this change?

**Comment by kkdd** [comment]
I have no idea about why the following downloading failed in building:

==> Downloading https://linuxbrew.bintray.com/bottles/berkeley-db-18.1.32_1.x86_64_linux.bottle.tar.gz
==> Downloading from https://akamai.bintray.com/a6/a61c738d8e2597fe1e662c1eda22005d1c846b2dee3516c89506c16d09fd0012?__gda__=exp=1587183110~hmac=77c4630145c4ce54cc227774ec19ab12b2c0502b292886b8401b3625550c909b&response-content-disposition=attachment%3Bfilename%3D%22berkeley-db-18.1.32_1.x86_64_linux.bottle.tar.gz%22&response-content-type=application%2Fgzip&requestInfo=U2FsdGVkX19NC90tR-ck_nnJkGj2-1svUG3GX2PCcFHj4W98aZDc16QiA2L6Rr7iHYOhMmAag9pAYZyMUsieD8OTn-Rj779J7XxF_aG4anLWKvuqLWE8B3tEJ3WrE-WVSarhAbetAxgaGUpsOoGVB9u499Ca82D7kryDaNvOAYk&response-X-Checksum-Sha1=b0a8bbc4d2b5b6b82e7a574cabe3a990006c4f5b&response-X-Checksum-Sha2=a61c738d8e2597fe1e662c1eda22005d1c846b2dee3516c89506c16d09fd0012
curl: (18) transfer closed with 20400624 bytes remaining to read
==> Retrying download
==> Downloading https://linuxbrew.bintray.com/bottles/berkeley-db-18.1.32_1.x86_64_linux.bottle.tar.gz
==> Downloading from https://akamai.bintray.com/a6/a61c738d8e2597fe1e662c1eda22005d1c846b2dee3516c89506c16d09fd0012?__gda__=exp=1587183110~hmac=77c4630145c4ce54cc227774ec19ab12b2c0502b292886b8401b3625550c909b&response-content-disposition=attachment%3Bfilename%3D%22berkeley-db-18.1.32_1.x86_64_linux.bottle.tar.gz%22&response-content-type=application%2Fgzip&requestInfo=U2FsdGVkX19NC90tR-ck_nnJkGj2-1svUG3GX2PCcFHj4W98aZDc16QiA2L6Rr7iHYOhMmAag9pAYZyMUsieD8OTn-Rj779J7XxF_aG4anLWKvuqLWE8B3tEJ3WrE-WVSarhAbetAxgaGUpsOoGVB9u499Ca82D7kryDaNvOAYk&response-X-Checksum-Sha1=b0a8bbc4d2b5b6b82e7a574cabe3a990006c4f5b&response-X-Checksum-Sha2=a61c738d8e2597fe1e662c1eda22005d1c846b2dee3516c89506c16d09fd0012
curl: (18) transfer closed with 16359808 bytes remaining to read
Error: Failed to download resource "berkeley-db"
Download failed: https://linuxbrew.bintray.com/bottles/berkeley-db-18.1.32_1.x86_64_linux.bottle.tar.gz
==> FAILED

File: Formula/osmfilter.rb (hunk 1 above)
Comment (kkdd) [reply Formula/osmfilter.rb]: Thank you. I will try it.

File: Formula/osmfilter.rb (hunk 2 above)
Comment (kkdd) [reply Formula/osmfilter.rb:25]: AD.tar.bz2 is used in the test and downloaded from [Gisgraphy download server](https://download.gisgraphy.com/openstreetmap/pbf/):
>  AD.tar.bz2  1.6 MB  (last update : June 13 2019 )

$ wget https://download.gisgraphy.com/openstreetmap/pbf/AD.tar.bz2
$ shasum -a 256 AD.tar.bz2
f8decd915758139e8bff2fdae6102efa0dc695b9d1d64cc89a090a91576efda9  AD.tar.bz2

Comment (iMichka) [reply Formula/osmfilter.rb:25]: If the shasum has changed, this means the upstream file has changed. We need clarification from upstream about this, and make sure no malicious thing has happened with this file.

The issue might be that they did not version their file ... so we have no clue when it is being changed under us.

**Comment by iMichka** [comment]
Regarding curl/download errors, you can re-trigger the build. It's probably just some transient network error.

File: Formula/osmfilter.rb (hunk 2 above)
Comment (kkdd) [reply Formula/osmfilter.rb]: I have changed the test as follows:

```
  test do
    system bin/"osmconvert", "-h"
  end
```

**Comment by kkdd** [comment]
What kind of conflict exists in the above "This branch has conflicts that must be resolved"?

**Comment by iMichka** [comment]
The version has probably been updated. You need to rebase this PR.

**Comment by iMichka** [comment]
I see that you actually made the changes in the Mac repo: https://github.com/Homebrew/homebrew-core/pull/52856

These changes have been merged and a linux bottle has been built:
https://github.com/Homebrew/linuxbrew-core/blob/master/Formula/osmfilter.rb

So this PR is useless as all the fixes are now in the linux repo.

On day both repos will be merged, so this back and forth will not be needed anymore.

Thanks for the fix!

**Pull request closed without merging**